lib/rules/no-self-assign.js

```diff
@@ -63,6 +63,9 @@ function isSameMember(left, right) {
     if (lobj.type === "MemberExpression") {
         return isSameMember(lobj, robj);
     }
+    if (lobj.type === "ThisExpression" && lobj.type === robj.type) {
```

Comment:
`lobj.type === robj.type` seems unnecessary as there is already early return above when types don't match?

Reply:
Indeed. I updated the PR